### DISCUSSION TITLE
How to add refunds?

### DISCUSSION THREAD

**Discussion by GurjinderSingh** [firefly-iii/firefly-iii#4828] [Firefly III]
I am running Firefly III  Version 5.5.11 

I was charged twice by some company ( Expense Account)
result in two transaction of same value for Example 20 $ = 20*2 = 40
then they refunded 20 $

Now, Expense Account show 40 $ spent and I have 20 $ transaction with destination to credit card(asset account) but no source .

my question is how do I decrease 20$ from Expense Account and credit asset account in one transaction.

another example is this correction transaction. I think we should have transaction type refund with source from expense account to asset account.

![image](https://user-images.githubusercontent.com/3470709/120557395-d5f03a00-c3cb-11eb-871d-d0a379e7eab3.png)

only way around I found is create a refunds account.

![image](https://user-images.githubusercontent.com/3470709/120557964-b9a0cd00-c3cc-11eb-9b8c-345fe5a18573.png)

**Comment by JC5** [comment]
You can also delete the second expense. A refund will never influence the expense account, it will always come from a revenue account.

**Reply by GurjinderSingh** [reply]
yes, that's possible only problem is when I return partial order so they refund half. For example, bought product worth 100 $ then returned 40 $ worth stuff in this case can't delete and return are possible till 90 days. plus in future while checking some transaction missing or non matching statements can confuse. 


thank you for your time and great software.

**Reply by Dinth** [reply]
Im also just started with Firefly III and im thinking how to handle that. Wont modifying/deleting one of the imported transactions cause it to re-import on the next import?

**Reply by JC5** [reply]
No, it won't.

**Reply by Jihad** [reply]
> Im also just started with Firefly III and im thinking how to handle that. Wont modifying/deleting one of the imported transactions cause it to re-import on the next import?

I'm going with making a deposit to CC account from the "merchant" revenue account. Not sure accounting-wise if this makes sense, but i can live with it for now.

**Comment by hoshsadiq** [comment]
I'm only just getting started with Firefly III and I just had a look around in my playground instance, and the way I found to be most intuitive is add a revenue account for the retail, e.g. "Costco refunds". Then just do a deposit from there. Tag it with a refund and then later add a link (there's some default links of which one is "partially refunded by").

Though, it would have been nicer to set the source account to be the expense account. @JC5 any reason not to allow the source account to be a expense account?

**Reply by JC5** [reply]
There's no reason not to give them the same name, but they'll have different ID's.

**Reply by hoshsadiq** [reply]
Understood, but it feels like setting the expense account as the source account would more accurately explain that this is a refund from a specific merchant.

**Reply by JC5** [reply]
It's an accountancy thing, to keep all of those accounts separately.

**Reply by hoshsadiq** [reply]
Sorry I don't think I'm following. Can you expand more?

I think a refund is a perfectly valid transaction as it should not not be counted as revenue. Revenue would count towards taxation in certain cases, whereas a refund would not. The link that can be created between two transactions is great, but in my opinion this isn't enough.

**Reply by JC5** [reply]
Revenue accounts and expense account are separate accounts to support double-entry bookkeeping. The first only receives money, the second one only spends it. This helps to keep internal consistency correct. 

Refunds will not be a separate transaction type. You can choose to either edit the original transaction, or to keep both transactions. It's up to you. I built it like this because the expense is done, the money is gone. If you get the money back for some reason that's a nice bonus, but since you budgeted the money and spent it, it remains spent. If you want to spend it "again", it's extra revenue that you can add to a budget or money you can funnel to your savings account to help you out next month.

**Reply by hoshsadiq** [reply]
I see what you mean. I suppose it's not always the case, sometimes it can be mistaken transactions, in which case it was never budgeted in the first place. Perhaps editing the transaction to subtract the refund in those scenarios is the right solution, because it does make sense in the scenario you mention. In those case I guess I can use tags to filter out transactions for tax purposes.

Thanks for the clarification!

**Reply by pogora** [reply]
I am new to Firefly and a bit late to the party. Still gotta give my 2 cents in that regard, since I also feel it should be possible to make +Positive transactions in order to refund things. 

Since double-entry bookkeeping was mentioned here a fun fact. In proper double entry bookkeeping you do not and cannot delete entries. Once they are booked they are booked. Thus if you make a mistake you have to create a canceling/storno entry. (that one would be on the other side and thus positive)

Maybe you could reconsider this long term and add the functionality? Pretty please? ;)

**Reply by GurjinderSingh** [reply]
right now I add extra transaction for refund, I do have few extra revenue accounts. its not that hard for me because I have received refund from like 6 companies only. most companies ask me if I want money back or settle next month. I am always settle next month guy or company gift cards :P.

**Reply by cachho** [reply]
> I am new to Firefly and a bit late to the party. Still gotta give my 2 cents in that regard, since I also feel it should be possible to make +Positive transactions in order to refund things.
> 
> Since double-entry bookkeeping was mentioned here a fun fact. In proper double entry bookkeeping you do not and cannot delete entries. Once they are booked they are booked. Thus if you make a mistake you have to create a canceling/storno entry. (that one would be on the other side and thus positive)
> 
> Maybe you could reconsider this long term and add the functionality? Pretty please? ;)



> I am new to Firefly and a bit late to the party. Still gotta give my 2 cents in that regard, since I also feel it should be possible to make +Positive transactions in order to refund things.
> 
> Since double-entry bookkeeping was mentioned here a fun fact. In proper double entry bookkeeping you do not and cannot delete entries. Once they are booked they are booked. Thus if you make a mistake you have to create a canceling/storno entry. (that one would be on the other side and thus positive)
> 
> Maybe you could reconsider this long term and add the functionality? Pretty please? ;)

I keep coming back to this issue and I agree. That's the way I learned it too. I think there should be a "pro" mode, for people who use it for more than their casual expense tracking, which allows all accounts to be used in expense and revenue transactions.

**Reply by Jihad** [reply]
> 
> I keep coming back to this issue and I agree. That's the way I learned it too. I think there should be a "pro" mode, for people who use it for more than their casual expense tracking, which allows all accounts to be used in expense and revenue transactions.

Yeah me too. I just started playing with FireFly for my household expenses. Wanted to add 6Months back of Credit Card Transactions (trying via API). I have lots of full & partial refunds on my CC, registered as Credit from same merchant.

I'm stuck not sure what to do with them, throws an error if i tried to either make it a negative, or if i "reversed" From/To destinations as well.

My Plan currently is to make every Negative/Credit amount in CC, as deposit. And FFIII will auto create revenue account for each automatically.

I wish I could link Merchants Expense & Revenue Accounts somehow, so i can trace that this X Merchant have those amounts from me and those amounts to me.

**Comment by Ezwen** [comment]
It seems I found the right thread to post my problem. I would like in Firefly to manage the following scenario:

- I have an asset account "Bank account"
- I have an expense account called "Taxes"
- I pay my taxes, therefore I create a transaction from "Bank account" to "Taxes"
- Later, I obtain a *tax refund* because of a calculation error. 

My first idea would be to create a reverse transaction from "Taxes" towards "Bank account". This would be ideal because (1) the refund would simply be a transaction in the other way around (easy to understand), and (2) it would make the "Taxes" account really reflect the final amount I did end up paying (considering the refund).

Unfortunately, Firefly does not seem to allow creating a transaction from an expense account towards an asset account. Therefore my only solution so far was to create a new revenue account called "Tax refund", which I find less practical than the ideal solution I described above.

All in all: I support the following idea stated in a previous message, which is to support creating transactions between all types of accounts :) (maybe through a toggle in the admin panel perhaps).

best wishes and thanks!

**Comment by myshevchuk** [comment]
The "Transfer" type of transaction in fact allows for proper double-entry debit/credit records within "Asset" accounts, where each side can be positive as well as negative. My solution was to create a special "Refunds" asset account to catch such entries. This way the transaction history is not altered while the balance reports are not affected by ghost amounts. Also, a refund _is_ an asset in a broader sense. It's just different from cash or checking account.
